### ISSUE TITLE
System.AccessViolationException Error when using TorchSharp

### ISSUE THREAD

**Issue by rapid-18** [dotnet/TorchSharp#1292]
A System.AccessViolationException Error occurs when executing a simple code
 model = jit.load(modelPath, DeviceType.CUDA);
how can I fit that
the torchsharp version is TorchSharp-cuda-windows 0.100.7
**the details of error information:**
System.AccessViolationException
  HResult=0x80004003
  Message=尝试读取或写入受保护的内存。这通常指示其他内存已损坏。
  Source=<Cannot evaluate the exception source>
  StackTrace:
<Cannot evaluate the exception stack trace>

**Comment by yueyinqiu** [comment]
It could work for me (the exception is not thrown by `jit.load`):

![image](https://github.com/dotnet/TorchSharp/assets/18749772/a6ec3df3-6aab-47f1-b79b-10392d214b00)

Could you please provide us more details, like the specific model file?

**Comment by rapid-18** [comment]
I could hardly provide any more details because it's just a simple pytorch trained model, and this problem also occurs in many other executions related to loading model, like below 
**nn.Module Model = torchvision.models.resnet50();**
so I think its not the problem from the model

**Comment by NiklasGustafsson** [comment]
Yes, please start by upgrading to the most recent version of TorchSharp -- it's very hard (let's call it impossible), given limited resources, for us to troubleshoot a version as old as `0.100.7`, which is based on an earlier version of libtorch.

**Comment by rapid-18** [comment]
This problem still exists even when I change to version 0.102.4 when loading models.......

**Comment by NiklasGustafsson** [comment]
Okay, that's good to know, that makes it easier to troubleshoot.

Can you please show the Python code used to generate the "exported.method.dat" file?

**Comment by yueyinqiu** [comment]
@NiklasGustafsson The `exported.method.dat` is created by me, not @rapid-18 .

And it could work for me. The exception is just I haven't use it correctly. Sorry for the misleading screenshot.

**Comment by travisjj** [comment]
This type of exception appears to be caused by the user's system running out of memory. 

As the code being executed is using pointers, the OOM exception is sort of vaguely just represented as a memory access violation. I have found expanding the virtual memory can alleviate this (for users with SSD's), however, if your model is excessively large, then you may consider trialing a small version to verify it runs and then porting it to a cloud GPU.

The memory issue can be verified by slowly expanding the system memory, and observing that the AccessViolation will occur in different places in code.

The core issue of using too much memory could happen from a wide variety of issues.

**Comment by alinpahontu2912** [comment]
Closing this issue due to lack of response, but feel free to reopen if needed
